src/jobwatcher/plugins/slurm.py

```diff
@@ -28,13 +28,27 @@ def get_required_nodes(instance_properties, max_size):
     )
     logging.info("Found the following pending jobs:\n%s", pending_jobs)
 
-    slots_requested = []
+    resources_requested = []
     nodes_requested = []
     for job in pending_jobs:
-        slots_requested.append(job.cpus_total)
+        resources_for_job = {}
```

Comment:
Propose to keep using dict, because we will be iterating through fields in resources, and I think looping using a dict makes the code easier to understand